File: src/main/python/linear_search.py
# REF: Linear Search Implementations -  https://teamtreehouse.com/library/introduction-to-algorithms/algorithms-in-code/linear-search-implementations


def linear_search(list, target):
    """
    Returns the index position of the target in the list if found, else None
    """

    # Loops worst case is Linear Time
    for i in range(0, len(list)):
        if list[i] == target:
            return i

    return None


def verify(index):
    if index is not None:
        print(f"Target found at index: {index}")
    else:
        print("Target not found in list")


# numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
numbers = [i for i in range(1, 10)]

result = linear_search(numbers, 12)
verify(result)

result = linear_search(numbers, 6)
verify(result)

File: src/main/javascript/algo/search/linear_search.js
function linearSearch(array, key) {
  // loops through the entire array and finds the first instance of the key
  for (let i = 0; i < array.length; i++) {
    if (array[i] === key) {
      return i;
    }
  }

  return -1;
}

function verify(index) {
  if (index === -1) {
    console.log('key not found in array');
  } else {
    console.log(`Key found at index: ${index}`);
  }
}

let result;
const numberArr = [1, 2, 3, 4, 5, 6];

result = linearSearch(numberArr, 8);
verify(result);

result = linearSearch(numberArr, 5);
verify(result);
